components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/DateTimePicker.jsx

```diff
@@ -78,49 +78,125 @@ export default class DateTimePicker extends React.Component {
         } else {
             days = 30;
         }
-
-        for (let i = 1; i <= days; i++) {
-            dayComponents.push(
-                <MenuItem
-                    key={`$days-${i}`}
-                    value={i}
-                    primaryText={i}
-                />
-            );
+        if(inputName === 'startTime') {
+            for (let i = 1; i <= days; i++) {
+                dayComponents.push(
```

Comment:
can we use the map instead of for loop and pushing array?

Reply:
Several more places in this file with similar case

Reply:
The value of variable `days` is an integer. It is not an array, therefore we can't use `.map()`.

Reply:
noted

Reply:
Fixed for variable `monthArray` with https://github.com/wso2/analytics-solutions/pull/306